draft-ietf-rtcweb-jsep.xml

```diff
@@ -1977,13 +1977,224 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
+        or ignore the value and move on. The implementation MAY store ignored
+        lines for diagnostic purposes or other reasons, but the data contained
+        within will not be used in any future JSEP processing. If the line is
+        not well-formed, or cannot be parsed as described, the parser MUST stop
+        with an error and reject the session decription. This ensures that
+        implementations do not accidentally misinterpret ambiguous SDP.</t>
+
+        <section title="Session-Level Parsing">
+          <t>First, the session-level lines are checked and parsed. These
+            lines MUST occur in a specific order, and with a specific syntax, as
+            defined in <xref target="RFC4566"/>, Section 5. Note that while
+            the specific line types (e.g. "v=", "c=") MUST occur in the defined
+            order, lines of the same type (typically "a=") can occur in any order,
+            and their ordering is not meaningful.</t>
+          <t>For non-attribute (non-"a=") lines, their sequencing and syntax are checked,
+            as mentioned above, but all values MUST be discarded. For example,
+            because the use of ICE is mandatory, the "c=" line is superfluous and its
+            value MUST be discarded; similarly, because session-level bandwidth is
+            ambiguous when multiple media streams are present, a "b=" line at session
+            level is not useful and its value MUST be discarded.</t>
+          <t>Specific processing is applied for the following session-level
+            attribute ("a=") lines:
+            <list style="symbols">
+              <t>Any "a=group" lines are parsed as specified in
+                <xref target="RFC5888"/>, Section 5, and, if
+                valid, the group's semantics and mids are stored.</t>
+               <t>If present, an "a=ice-lite" line is parsed as specified in
+                <xref target="RFC5245"/>, Section 15.3, and, if valid, a value indicating the presence of ice-lite is stored.</t>
+              <t>If present, an "a=ice-ufrag" line is parsed as specified in
+                <xref target="RFC5245"/>, Section 15.4, and, if valid,
+                the ufrag value is stored.</t>
+              <t>If present, an "a=ice-pwd" line is parsed as specified in
+                <xref target="RFC5245"/>, Section 15.4, and, if valid, the pwd
+                value is stored.</t>
+              <t>If present, an "a=ice-options" line is parsed as specified in
+                <xref target="RFC5245"/>, Section 15.5, and, if valid, the set of specified options is stored.</t>
+              <t>If present, any "a=fingerprint" lines are parsed as specified in
+                <xref target="RFC4572"/>, Section 5, and, if valid, the set of fingerprint and algorithm values is stored.</t>
+              <t>If present, an "a=setup" line is parsed as specified in
+                <xref target="RFC4145"/>, Section 4, and, if valid, the value
```

Comment:
We could probably be terser here and say "read the ICE ufrag, pwd, options, ice-lite" as defined in 5245, Sectiond 15.3-15.5. I do think we need to be prescriptive here since we are going to later use these parsed values in the apply section, and it needs to be clear where they came from.